### PR TITLE
refactor: setup histoire and workspaces

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### What's new?

- [x] at first https://github.com/kodadot/nft-gallery/commit/bd3127f17a42c3d35bc5626d4f0a2a70be23aa0e, I want to move our nuxt app under the folder. but, to minimize the changes and we don't need turborepo right now. I decided to setup workspaces and histoire only. so, we can start to using it for new redesign page
- [x] setup pnpm workspaces
- [x] setup histoire
- [x] please put newer component under `@kodadot/ui`
- [x] a little example in `components/landing/FeaturedArticles.vue`

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by vikiival** [comment]
Looking forward

File: .vscode/vetur/snippets/template-typescript-w-decorator.vue [hunk 1]
```diff
@@ -1,13 +0,0 @@
-<template>
```
Comment (vikiival) [inline .vscode/vetur/snippets/template-typescript-w-decorator.vue:1]: maybe would be good to have snippet for composition API?

File: libs/ui/src/scss/variable.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+$k-accent: #ff7ac3;
```
Comment (vikiival) [inline libs/ui/src/scss/variable.scss:1]: Is this from figma file? cc @exezbcz

File: package.json [hunk 3]
```diff
@@ -66,6 +66,7 @@
     "@fortawesome/vue-fontawesome": "^2.0.8",
     "@google/model-viewer": "^1.12.1",
     "@keeex/qrcodejs-kx": "^1.0.2",
+    "@kodadot/ui": "workspace:*",
```
Comment (vikiival) [inline package.json:69]: thinking to have cooler name for that 🤔 may @exezbcz/ @yangwao have some ideas?

File: libs/ui/src/scss/variable.scss (hunk 2 above)
Comment (exezbcz) [reply libs/ui/src/scss/variable.scss:1]: yes sir
image

**Review by vikiival — COMMENTED** [review]
Smol stuff

**Review by roiLeo — COMMENTED** [review]
Looks nice!
Capture d’écran 2022-10-24 à 4 09 22 PM

- it miss to link somewhere histoire doc 
- should we had lorem ipsum instead of actual content?
- can we handle Light/Dark mode variant globally? how is it going to work with multiple buttons variant?

File: .vscode/vetur/snippets/template-typescript-w-decorator.vue (hunk 1 above)
Comment (preschian) [reply .vscode/vetur/snippets/template-typescript-w-decorator.vue:1]: good point. maybe we need to list recommended extensions in the readme files. this extension seems good already https://marketplace.visualstudio.com/items?itemName=sdras.vue-vscode-snippets. any idea where to put that list?

**Comment by preschian** [comment]
> * it miss to link somewhere histoire doc
> * should we had lorem ipsum instead of actual content?

aahh yes, this is important. I forgot to add histoire doc link in the readme

> * can we handle Light/Dark mode variant globally? how is it going to work with multiple buttons variant?

need to find a workaround for light/dark mode globally

File: .vscode/vetur/snippets/template-typescript-w-decorator.vue (hunk 1 above)
Comment (preschian) [reply .vscode/vetur/snippets/template-typescript-w-decorator.vue:1]: I put recommended extensions in here https://github.com/preschian/nft-gallery/blob/init-histoire-and-workspaces/CONTRIBUTING.md#getting-started

**Review by roiLeo — APPROVED** [review]
I look forward to working with histoire

File: package.json (hunk 3 above)
Comment (yangwao) [reply package.json:69]: something which is more contextual yes

**Comment by vikiival** [comment]
@preschian we were talking with @yangwao to rename the library to `@kodadot/brick`

- because we are making building blocks
- and should evoke the neobrutalism style
- + it is easy to pronounce

also maybe one thing: on npm we have org `@kodadot1`

**Comment by preschian** [comment]
> @preschian we were talking with @yangwao to rename the library to `@kodadot/brick`
> 
> * because we are making building blocks
> * and should evoke the neobrutalism style
> * * it is easy to pronounce
> 
> also maybe one thing: on npm we have org `@kodadot1`

noted, will rename it 👍

**Review by vikiival — APPROVED** [review]

**Comment by yangwao** [comment]
conflictz :|

**Review by yangwao — APPROVED** [review]

**Pull request merged**